mso/resource_mso_schema_site_anp_epg_domain.go

```diff
@@ -58,18 +58,55 @@ func resourceMSOSchemaSiteAnpEpgDomain() *schema.Resource {
 				ForceNew:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
 			},
-
-			"domain_type": &schema.Schema{
+			"domain_name": &schema.Schema{
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				ForceNew:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
 			},
-			"dn": &schema.Schema{
+			"domain_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"vmmDomain",
+					"l3ExtDomain",
+					"l2ExtDomain",
+					"physicalDomain",
+					"fibreChannelDomain",
+				}, false),
+			},
+			"vmm_domain_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"VMware",
+					"Microsoft",
+					"Redhat",
+				}, false),
+			},
+			"domain_dn": &schema.Schema{
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				ForceNew:     true,
 				ValidateFunc: validation.StringLenBetween(1, 1000),
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if new == "" {
+						return true
+					} else {
+						return false
+					}
+
+				},
```

Comment:
What is this used for?

Reply:
This is used to retain the dn in the state regardless of whether it was entered by the user or not.